pkg/cloud/rgraph/exec/retrieable_action.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2024 Google LLC
+
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package exec
+
+import (
+	"context"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+)
+
+// retriableAction is a decorator to the action adding retry mechanism
+type retriableAction struct {
+	Action
+	isRetriable func(error) bool
+}
+
+// NewRetriableAction decorates Action with retry provider function
```

Comment:
This comment docstring doesn't describe what the `rp` parameter means. 

Please always write the public docstrings in a way that someone reading the godoc page will be able to use the package without reading the actual underlying implementation. It also serves as a guide for defining the semantics of function.

Reply:
done